cmd/inventory/utils_openstack.go

```diff
@@ -0,0 +1,461 @@
+// SPDX-FileCopyrightText: 2025 SAP SE or an SAP affiliate company and Gardener contributors
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package main
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"log/slog"
+	"os"
+	"strings"
+
+	openstackclients "github.com/gardener/inventory/pkg/clients/openstack"
+	"github.com/gardener/inventory/pkg/core/config"
+
+	"github.com/gophercloud/gophercloud/v2"
+	"github.com/gophercloud/gophercloud/v2/openstack"
+	gophercloudconfig "github.com/gophercloud/gophercloud/v2/openstack/config"
+)
+
+var errNoUsernameFile = errors.New("no username file specified")
+var errNoPasswordFile = errors.New("no password file specified")
+var errNoAppCredentialsIDFile = errors.New("no app credentials id file specified")
+var errNoAppCredentialsSecretFile = errors.New("no app credentials secret file specified")
+var errNoAppCredentialsNameFile = errors.New("no app credentials name file specified")
+var errNoAuthEndpoint = errors.New("no authentication endpoint specified")
+var errNoDomain = errors.New("no domain specified")
+var errNoRegion = errors.New("no region specified")
+var errNoProject = errors.New("no project specified")
+var errNoProjectID = errors.New("no project id specified")
+
+// validateOpenStackConfig validates the OpenStack configuration settings.
+func validateOpenStackConfig(conf *config.Config) error {
+	// Make sure that the services have named credentials configured.
+	services := map[string][]config.OpenStackServiceConfig{
+		"compute": conf.OpenStack.Services.Compute,
+		"network": conf.OpenStack.Services.Network,
+		"storage": conf.OpenStack.Services.BlockStorage,
```

Comment:
Probably should be `block_storage` instead of the generic `storage`, since we have `object_storage` as well.